### ISSUE TITLE
Add a PrimaryAttribute type

### ISSUE THREAD

**Issue by TylerBrinkley** [dotnet/runtime#28200]
Split from dotnet/runtime#20008 

For the `Enum` method `GetName` when there are multiple enum members with the same value the name it returns is undefined. I propose adding a `PrimaryAttribute` to be applied to enum members where there is ambiguity which will ensure its name is returned.

## Proposed API
```c#
namespace System.ComponentModel {
    public sealed class PrimaryAttribute : Attribute {
        public PrimaryAttribute();
    }
}
```

## API Details
* The `PrimaryAttribute` is used to specify what enum member to retrieve when there are multiple enum members with the same value. It could be used for more than just enum members for cases in code where there is ambiguity.

**Comment by ericstj** [comment]
Though the API as proposed places this in System.ComponentModel, it would need to be understood by Enum.GetName which is actually CoreLib / System.Runtime.

FWIW it isn't "undefined" it's just not entirely predictable.

**Comment by TylerBrinkley** [comment]
Yes, it would need to be defined in corelib to be understood by System.Enum.

It could additionally be used for specifying the constructor to invoke in deserialization.
